File: src/dotnet/FaxCore.ApiSamples/Samples/User/Contact/CreateContact.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Contact
{
    internal static class CreateContact
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            RequireDestructiveConfirmation();

            return client.PostAsync("/api/contact", new
            {
                addressBookID = config.AddressBookID,
                displayName = config.ContactDisplayName,
                firstName = config.ContactFirstName,
                lastName = config.ContactLastName,
                compName = config.ContactCompany,
                preferAddressType = config.ContactPreferAddressType,
                visibility = config.ContactVisibility,
                description = config.ContactDescription,
                notifyOnFailed = config.ContactNotifyOnFailed,
                notifyOnSuccess = config.ContactNotifyOnSuccess,
                addressList = new[]
                {
                    new
                    {
                        addrType = config.ContactAddressType,
                        address = config.ContactAddress
                    }
                }
            });
        }

        private static void RequireDestructiveConfirmation()
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to create a contact.");
            }
        }
    }
}

File: src/javascript/user/contact/create-contact.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to create a contact.');
}

const client = new FaxCoreClient();
const result = await client.post('/api/contact', {
  addressBookID: Number(process.env.FAXCORE_ADDRESS_BOOK_ID || 0),
  displayName: process.env.FAXCORE_CONTACT_DISPLAY_NAME || 'Sample Contact',
  firstName: process.env.FAXCORE_CONTACT_FIRST_NAME || 'Sample',
  lastName: process.env.FAXCORE_CONTACT_LAST_NAME || 'Contact',
  compName: process.env.FAXCORE_CONTACT_COMPANY || 'Example Company',
  preferAddressType: process.env.FAXCORE_CONTACT_PREFER_ADDRESS_TYPE || 'Fax',
  visibility: process.env.FAXCORE_CONTACT_VISIBILITY || 'Private',
  description: process.env.FAXCORE_CONTACT_DESCRIPTION || 'Created from JavaScript sample',
  notifyOnFailed: (process.env.FAXCORE_CONTACT_NOTIFY_ON_FAILED || 'true').toLowerCase() !== 'false',
  notifyOnSuccess: (process.env.FAXCORE_CONTACT_NOTIFY_ON_SUCCESS || 'true').toLowerCase() !== 'false',
  addressList: [
    {
      addrType: process.env.FAXCORE_CONTACT_ADDRESS_TYPE || 'Fax',
      address: process.env.FAXCORE_CONTACT_ADDRESS || '+15551234567'
    }
  ]
});

printResult('Create contact', result);
